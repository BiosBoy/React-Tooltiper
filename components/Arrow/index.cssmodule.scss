.arrow {
  z-index: 100000;
}

.arrowlefttop,
.arrowrighttop,
.arrowcentercenter,
.arrowrightbottom,
.arrowleftbottom {
  display: none;
}

.arrowrightcenter {
  transform: rotate(90deg);
}

.arrowleftcenter {
  transform: rotate(-90deg);
}

.arrowcenterbottom {
  transform: rotate(180deg);
}

.arrowcentertop {
  transform: rotate(0deg);
}
