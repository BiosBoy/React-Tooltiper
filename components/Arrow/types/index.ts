import { IArrow } from '../../../types'

export interface IProps extends IArrow {
  arrowWrapStyles?: string
  arrowStyles?: string
  arrowRotate?: {
    coordX: string
    coordY: string
  }
}
