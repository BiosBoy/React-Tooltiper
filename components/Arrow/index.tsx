import React, { PureComponent } from 'react'

import { IProps } from './types'
import styles from './index.cssmodule.scss'

class Arrow extends PureComponent<IProps> {
  render() {
    const {
      addArrow,
      arrowWrapStyles,
      arrowStyles,
      arrowRotate: { coordX = 'center', coordY = 'top' } = {}
    } = this.props

    if (!addArrow) {
      return null
    }

    return (
      <div className={arrowWrapStyles}>
        <div className={`${styles.arrow} ${styles[`arrow${coordX}${coordY}`]} ${arrowStyles}`} />
      </div>
    )
  }
}

export default Arrow
