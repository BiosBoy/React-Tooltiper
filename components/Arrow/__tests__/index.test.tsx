import React from 'react'
import { mount } from 'enzyme'

import initialState, { disabled, rotatedArrow } from './mocks'
import Arrow from '..'

describe('<Tooltip Arrow test />', () => {
  it('should render basic Arrow', () => {
    const Component = mount(<Arrow {...initialState} />)

    expect(Component.find('Arrow').length).toBe(1)
    expect(Component.find('.arrowWrap').length).toBe(1)
    expect(Component.find('.tooltipArrow').length).toBe(1)

    expect(Component).toMatchSnapshot()
  })
  it('should not render Arrow in case of addArrow = false flag ', () => {
    const Component = mount(<Arrow {...disabled} />)

    expect(Component.find('Arrow').length).toBe(1)
    expect(Component.find('.arrowWrap').length).toBe(0)
    expect(Component.find('.tooltipArrow').length).toBe(0)

    expect(Component).toMatchSnapshot()
  })
  it('should render rotated Arrow on 180 degree', () => {
    const Component = mount(<Arrow {...rotatedArrow} />)

    expect(Component.find('Arrow').length).toBe(1)
    expect(Component.find('.arrowWrap').length).toBe(1)
    expect(Component.find('.tooltipArrow').length).toBe(1)
    expect(Component.find('.tooltipArrow').prop('className')).toContain('arrowcenterbottom')

    expect(Component).toMatchSnapshot()
  })
})
