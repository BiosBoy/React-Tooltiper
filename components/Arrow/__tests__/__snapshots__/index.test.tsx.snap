// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip Arrow test /> should not render Arrow in case of addArrow = false flag  1`] = `
<Arrow
  addArrow={false}
  arrowStyles="tooltipArrow"
  arrowWrapStyles="arrowWrap"
/>
`;

exports[`<Tooltip Arrow test /> should render basic Arrow 1`] = `
<Arrow
  addArrow={true}
  arrowStyles="tooltipArrow"
  arrowWrapStyles="arrowWrap"
>
  <div
    className="arrowWrap"
  >
    <div
      className="arrow arrowcentertop tooltipArrow"
    />
  </div>
</Arrow>
`;

exports[`<Tooltip Arrow test /> should render rotated Arrow on 180 degree 1`] = `
<Arrow
  addArrow={true}
  arrowRotate={
    Object {
      "coordX": "center",
      "coordY": "bottom",
    }
  }
  arrowStyles="tooltipArrow"
  arrowWrapStyles="arrowWrap"
>
  <div
    className="arrowWrap"
  >
    <div
      className="arrow arrowcenterbottom tooltipArrow"
    />
  </div>
</Arrow>
`;
